examples/demo.sh

```
@@ -0,0 +1,463 @@
+
+# Shell options and error handling
+set -euo pipefail
+IFS=$'\n\t'
+
+# Color definitions using ANSI escape codes
+readonly RED='\033[0;31m'
+readonly GREEN='\033[0;32m'
+readonly YELLOW='\033[0;33m'
+readonly BLUE='\033[0;34m'
+readonly PURPLE='\033[0;35m'
+readonly CYAN='\033[0;36m'
```

Shellcheck (0.10.0)

[warning] CYAN appears unused. Verify use (or export if used externally).

(SC2034)